src/langchain_google_firestore/document_loader.py

```diff
@@ -0,0 +1,199 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import itertools
+from typing import TYPE_CHECKING, Any, Iterator, List, Optional
+
+import more_itertools
+from google.cloud import firestore
+from google.cloud.firestore import DocumentReference
+from google.cloud.firestore_v1.services.firestore.transports.base import (
+    DEFAULT_CLIENT_INFO,
+)
+from langchain_community.document_loaders.base import BaseLoader
+from langchain_core.documents import Document
+
+from .utility.document_converter import DocumentConverter
+
+DEFAULT_FIRESTORE_DATABASE = "(default)"
+USER_AGENT = "langchain-google-firestore-python"
+WRITE_BATCH_SIZE = 500
+
+if TYPE_CHECKING:
+    from google.cloud.firestore import Client, CollectionGroup, DocumentReference, Query
+
+
+class FirestoreLoader(BaseLoader):
+    def __init__(
+        self,
+        source: Query | CollectionGroup | DocumentReference | str,
+        page_content_fields: List[str] = None,
+        metadata_fields: Optional[List[str]] = None,
+        client: Optional[Client] = None,
+    ) -> None:
+        """Document Loader for Google Cloud Firestore.
+        Args:
+            source: The source to load the documents. It can be an instance of Query,
+                CollectionGroup, DocumentReference or the single `/`-delimited path to
+                a Firestore collection.
+            page_content_fields: The document field names to write into the `page_content`.
+                If an empty or None list is provided all fields will be written into
+                `page_content`. When only one field is provided only the value is written.
+            metadata_fields: The document field names to write into the `metadata`.
+                By default it will write all fields that are not in `page_content` into `metadata`.
+            client: Client for interacting with the Google Cloud Firestore API.
+        """
+        if client:
+            self.client = client
+            self.client._user_agent = USER_AGENT
+        else:
+            client_info = DEFAULT_CLIENT_INFO
+            client_info.user_agent = USER_AGENT
+            self.client = firestore.Client(client_info=client_info)
+        self.source = source
+        self.page_content_fields = page_content_fields
+        self.metadata_fields = metadata_fields
+
+    def load(self) -> List[Document]:
+        """Load Documents."""
+        return list(self.lazy_load())
+
+    def lazy_load(self) -> Iterator[Document]:
+        """A lazy loader for Documents."""
+        if isinstance(self.source, DocumentReference):
+            self.source._client._client_info.user_agent = USER_AGENT
+            doc = self._load_document()
+            if doc:
+                yield self._load_document()
+            return
```

Comment:
Notice sometimes the file "return None" vs Just "return" in _load_document (also difference for type annotation). Consider applying the same standard.

Reply:
I changed to return an empty list []. 
Returning None is a non-iterable and would need more logic to handle it.

When I yield the self._load_document is then ok.  I tried just returning the document but had problems, and verified with simpler code I had to yield.